src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -1,60 +1,71 @@
 package org.trigon.hardware.rev.spark.io;
 
-import com.revrobotics.*;
-import edu.wpi.first.math.system.plant.DCMotor;
+import com.revrobotics.sim.SparkAbsoluteEncoderSim;
+import com.revrobotics.sim.SparkRelativeEncoderSim;
+import com.revrobotics.spark.SparkBase;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.SparkSim;
+import com.revrobotics.spark.config.SparkBaseConfig;
+import com.revrobotics.spark.config.SparkMaxConfig;
+import org.trigon.hardware.RobotHardwareStats;
 import org.trigon.hardware.rev.spark.SparkIO;
 import org.trigon.hardware.rev.sparkecnoder.SparkEncoder;
+import org.trigon.hardware.simulation.MotorPhysicsSimulation;
 
 public class SimulationSparkIO extends SparkIO {
-    private final CANSparkMax motor;
-    private final SparkPIDController pidController;
+    private final SparkMax motor;
+    private final SparkClosedLoopController pidController;
     private final SparkEncoder encoder;
+    private SparkSim motorSimulation = null;
+    private SparkAbsoluteEncoderSim absoluteEncoderSimulation = null;
+    private SparkRelativeEncoderSim relativeEncoderSimulation = null;
+    private MotorPhysicsSimulation physicsSimulation = null;
 
-    public SimulationSparkIO(int id, DCMotor gearbox) {
-        motor = new CANSparkMax(id, CANSparkMax.MotorType.kBrushless);
-        REVPhysicsSim.getInstance().addSparkMax(motor, gearbox);
-        pidController = motor.getPIDController();
-        encoder = SparkEncoder.createRelativeEncoder(motor);
+    public SimulationSparkIO(int id) {
+        motor = new SparkMax(id, SparkMax.MotorType.kBrushless);
+        pidController = motor.getClosedLoopController();
+        encoder = SparkEncoder.createEncoder(motor);
```

Comment:
Still doesn't solve anything, you want to determine the encoder from the variable you get in the sim function.